omegaconf/basecontainer.py

```diff
@@ -387,7 +387,7 @@ def _merge_with(
             if isinstance(self, DictConfig) and isinstance(other, DictConfig):
                 BaseContainer._map_merge(self, other)
             elif isinstance(self, ListConfig) and isinstance(other, ListConfig):
-                if self._is_none() or self._is_missing() or self._is_interpolation():
+                if self._is_none() or self._is_interpolation() or self._is_missing():
                     self.__dict__["_content"] = []
                 else:
                     self.__dict__["_content"].clear()
```

Comment:
This bit puzzles me. the two branches are pretty similar.
Can you try to see if you can just use one of the branches always?

Reply:
This block can be reduced to `self.__dict__["_content"] = []` without the if-else and works the same

Reply:
Cool, please do it.